lightning/src/ln/msgs.rs

```diff
@@ -59,6 +59,9 @@ pub enum DecodeError {
 	/// or 1, a public key/private key/signature was invalid, text wasn't UTF-8, TLV was
 	/// syntactically incorrect, etc
 	InvalidValue,
+	/// Same as [`InvalidValue`], however this is set for gossip messages where we may not want to
+	/// force-close our channels with a counterparty and can instead send a [`Warning`] message.
+	InvalidGossipValue,
```

Comment:
Hmm... this seems like the wrong layer for this. Can't the call site condition on the message type?

Reply:
I pushed an alternative implementation, let me know if you prefer this one.

Reply:
Thanks, I like the new version but may need to condition the warning based on the type of error?

Reply:
Not sure what you mean? The match is done in `peer_handler`?

Reply:
Yeah, the reordering change addresses my concern.